glib/Object.cs

```diff
@@ -92,13 +92,13 @@ public static Object GetObject(IntPtr o)
 
 		private static void ConnectDefaultHandlers (GType gtype, System.Type t)
 		{
-			object [] parms = new object [1];
 			foreach (MethodInfo minfo in t.GetMethods(BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.DeclaredOnly)) {
 				MethodInfo baseinfo = minfo.GetBaseDefinition ();
 				if (baseinfo == minfo)
 					continue;
 
 				foreach (object attr in baseinfo.GetCustomAttributes (typeof (DefaultSignalHandlerAttribute), false)) {
+					object [] parms = new object [1];
 					DefaultSignalHandlerAttribute sigattr = attr as DefaultSignalHandlerAttribute;
```

Comment:
You did this then undid it, maybe add a comment explaining why it can't be moved?

Reply:
Because of the break below, it makes no sense to pre-allocate when we should allocate when needed.